integration_tests/base_routes.py

```diff
@@ -823,33 +825,259 @@ def sync_router_di(request, router_dependencies):
+        "url": request_url.path,
+        "headers": request_headers.get("server"),
+    }
+
+
+@app.post("/async/split_request_typed/combined")
+async def async_split_request_combined(
+    request_body: Body,
+    query_data: QueryParams,
+    request_method: Method,
+    request_url: Url,
+    request_headers: Headers,
+):
+    return {
+        "body": request_body,
+        "query_params": query_data.to_dict(),
+        "method": request_method,
+        "url": request_url.path,
+        "headers": request_headers.get("server"),
+    }
+
+
+@app.post("/sync/split_request_typed_untyped/combined")
+def sync_split_request_typed_untyped_combined(
+    query_params,
+    request_method: Method,
+    request_body: Body,
+    url: Url,
+    headers: Headers,
+):
+    return {
+        "body": request_body,
+        "query_params": query_params.to_dict(),
+        "method": request_method,
+        "url": url.path,
+        "headers": headers.get("server"),
+    }
+
+
+@app.post("/async/split_request_typed_untyped/combined")
+async def async_split_request_typed_untyped_combined(
+    query_params,
+    request_method: Method,
+    request_body: Body,
+    url: Url,
+    headers: Headers,
+):
+    return {
+        "body": request_body,
+        "query_params": query_params.to_dict(),
+        "method": request_method,
+        "url": url.path,
+        "headers": headers.get("server"),
+    }
+
+
+@app.post("/sync/split_request_typed_untyped/combined/failure")
+def sync_split_request_typed_untyped_combined_failure(query_params, request_method: Method, request_body: Body, url: Url, headers: Headers, vishnu):
+    return {
+        "body": request_body,
+        "query_params": query_params.to_dict(),
+        "method": request_method,
+        "url": url.path,
+        "headers": headers.get("server"),
+        "vishnu": vishnu,
+    }
+
+
+@app.post("/async/split_request_typed_untyped/combined/failure")
+async def async_split_request_typed_untyped_combined_failure(query_params, request_method: Method, request_body: Body, url: Url, headers: Headers, vishnu):
+    return {
+        "body": request_body,
+        "query_params": query_params.to_dict(),
+        "method": request_method,
+        "url": url.path,
+        "headers": headers.get("server"),
+        "vishnu": vishnu,
+    }
+
+
 @app.get("/openapi_test", openapi_tags=["test tag"])
 def sample_openapi_endpoint():
     """Get openapi"""
     return 200
 
 
-class Initial(TypedDict):
+class Initial(Body):
     is_present: bool
     letter: Optional[str]
 
 
-class FullName(TypedDict):
+class FullName(Body):
     first: str
     second: str
     initial: Initial
 
 
-class CreateItemBody(TypedDict):
+class CreateItemBody(Body):
     name: FullName
     description: str
     price: float
     tax: float
 
 
+class CreateItemResponse(JSONResponse):
+    success: bool
+    items_changed: int
+
+
+class CreateItemQueryParamsParams(QueryParams):
+    required: bool
```

Comment:
if this work, does this imply we also don't need to have a new type alias for `Body` and other things(like `JSONResponse`)?

Reply:
yep, I'll do so

Reply:
@sansyrox yep, `JSONResponse` can be `robyn/Response`. but, what about `Body` -- we don't have a body object.

Reply:
@sansyrox nops, if we inherit `Body`, we would have to pass the mandatory params to it (`TypeError: Response.__new__() missing 3 required positional arguments: 'status_code', 'headers', and 'description'`). also, we don't have an object for `Body`.